lizmap/modules/lizmap/classes/qgisForm.class.php

```diff
@@ -1161,7 +1161,10 @@ private function fillControlFromValueRelationLayer($fieldName, $formControl)
             $dataSource = new jFormsStaticDatasource();
 
             // required
-            if (!$formControl->valueRelationData['allowNull']) {
+            if (array_key_exists('notNull', $formControl->valueRelationData)
+                and $formControl->valueRelationData['notNull']
+            ) {
+                jLog::log('notNull '.$formControl->valueRelationData['notNull'], 'error');
                 $formControl->ctrl->required = true;
             }
```

Comment:
I think the field is require when `notNull` is check and not `allowNull`

Reply:
Are you sure that `notNull` is in `valueRelationData` ?